### PR TITLE
Refactor/115 flask to fastapi

### PR DESCRIPTION
**Pull request by Branchverse**
- replace Flask with FastAPI
- Make APIKey global and use via OpenApi
- replace waitress with uvicorn because Flask was WSGI and FastAPI is ASGI
- set custom properties for OpenApi
- Refactor some code in the new app.py
- 

ToDo:

- [x] Fix default values for POST /transcriptions in swagger
- [x] Fix / refactor tests

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -12,3 +12,5 @@ stable_ts_whisperless==2.17.3
 ruff==0.1.15
 numpy<2
 ctranslate2==4.4.0
+fastapi==0.115.4
+python-multipart==0.0.17
```
Comment (coffeemakingtoaster) [inline requirements.txt:16]: Where is this used?

File: src/rest/app.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+from src.helper.config import CONFIG
+from src.helper.data_handler import DataHandler
+from src.helper.types.transcription_status import TranscriptionStatus
+
+LOGGER = logging.getLogger(__name__)
+DATA_HANDLER = DataHandler()
+
+
+class TranscriptionData(BaseModel):
```
Comment (coffeemakingtoaster) [inline src/rest/app.py:23]: Move this to a separate file maybe? I think this makes this file more convoluted than it needs to be.

File: src/rest/app.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    language: str | None = None
+
+
+config = CONFIG
+
+app = FastAPI()
+api_key_header = APIKeyHeader(name="Authorization", auto_error=False)
+
+
+def custom_openapi():
```
Comment (coffeemakingtoaster) [inline src/rest/app.py:40]: This is missing the "what does it do?" comment. 
I don't think they are necessary per se but present everywhere else...therefore it is great for unified code style

Comment (coffeemakingtoaster) [inline src/rest/app.py:40]:
```suggestion
def custom_openapi():
"""" Modify the openapi definition to include our auth schema """
```

File: src/rest/app.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    try:
+        audio = AudioSegment.from_file(file.file)
+        result = DATA_HANDLER.save_audio_file(audio, transcription_id)
+        if not result["success"]:
+            raise HTTPException(status_code=400, detail=result["message"])
+    except Exception as e:
+        LOGGER.error(f"Audio processing error: {e}")
+        raise HTTPException(status_code=500, detail="Something went wrong")
+
+    time.sleep(0.1)  # ensure file is saved
```
Comment (coffeemakingtoaster) [inline src/rest/app.py:135]: I still hate this
```suggestion
    time.sleep(0.1)  # TODO: Remove this line...
```

File: src/rest/app.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        start_time=datetime.utcnow().isoformat() + 'Z',
+        settings=settings_dict,
+        model=model,
+        task=task,
+        text=text,
+        language=language,
+    ).model_dump()
+
+    DATA_HANDLER.write_status_file(transcription_id, data)
+    return JSONResponse(content=data, status_code=200)
```
Comment (coffeemakingtoaster) [inline src/rest/app.py:157]: How long does this request take? This seems like it may be very slow

Not really relevant for this PR though

File: src/rest/app.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    if file:
+        return JSONResponse(content=file, status_code=200)
+    raise HTTPException(status_code=404, detail="Transcription ID not found")
+
+
+@app.get("/export/audio/{transcription_id}", dependencies=[Depends(require_api_key)])
+async def get_stream_audio_export(transcription_id: str):
+    """Get the audio WAV file for a specific transcription ID."""
+    file = DATA_HANDLER.get_audio_file_by_id(transcription_id)
+    if file is not None:
```
Comment (coffeemakingtoaster) [inline src/rest/app.py:173]: I think its prettier to just flip this.
Keyword: Guard Clauses

File: src/rest/run.py [hunk 7]
```diff
@@ -2,18 +2,20 @@
 
 import logging
 import multiprocessing
-import waitress
-from src.rest.runner import Runner
+
+import uvicorn
```
Comment (coffeemakingtoaster) [inline src/rest/run.py:6]: this is not in requirements.text...should it be?

**Review by coffeemakingtoaster — CHANGES_REQUESTED** [review]
Pre review 

Some actual change requests...some of them are just me rambling about stuff I don't linke

File: src/rest/run.py (hunk 7 above)
Comment (Branchverse) [reply src/rest/run.py:6]: huh, I did put it in I thought o.0
Thx for the good eyes

File: requirements.txt (hunk 1 above)
Comment (Branchverse) [reply requirements.txt:16]: These are usually required by one of the other imports, upon installation they say, this package also needs xyz installed without installing it as a dependancy itself so I included it

File: src/rest/app.py (hunk 4 above)
Comment (Branchverse) [reply src/rest/app.py:136]: agreed

File: src/rest/app.py (hunk 5 above)
Comment (Branchverse) [reply src/rest/app.py:158]: in other files are time_it decorators, we can go ahead and implement them everywhere to track statistics of the service itself if you would like?

File: src/rest/app.py (hunk 6 above)
Comment (Branchverse) [reply src/rest/app.py:174]: tbh, I had it flipped at first and found it ugly hahhaha

File: requirements.txt (hunk 1 above)
Comment (coffeemakingtoaster) [reply requirements.txt]: Okay if you want to maybe validate that our use case really needs it. The only other occurence of multipart was already in the code. 

If you dont want to maybe at that as a note to the requirements.txt or the readme

File: src/rest/app.py (hunk 5 above)
Comment (coffeemakingtoaster) [reply src/rest/app.py:163]: Lets leave it for now. Maybe in the long run we can make the decorator env dependant. Meaning we can toggle it on and off. Otherwise "instrumenting" every function is gonna bloat our logs. Also python may have a build in way to track function execution times on that scale

File: docs/rest-api.md [hunk 8]
```diff
@@ -49,7 +49,7 @@ Example:
 
 ```text
 Header: key: shuffle2024
-Body: settings: {"test": "test"}, model: "tiny", file: "test.wav"
+Body: settings: {"test": "test"}, model: "large-v3", file: "test.wav"
```
Comment (coffeemakingtoaster) [inline docs/rest-api.md:52]: Is this change needed?

File: src/rest/app.py (hunk 2 above)
Comment (coffeemakingtoaster) [reply src/rest/app.py:24]: this still stands

**Review by coffeemakingtoaster — COMMENTED** [review]
I like it. I havent tested it on my machine yet, will approve once I did that

File: docs/rest-api.md (hunk 8 above)
Comment (Branchverse) [reply docs/rest-api.md:52]: If you start the server it runs with the large-v3 and the tiny model example request won't work, so I changed the example for it to immediately work, wasn't that an issue for you?

File: requirements.txt (hunk 1 above)
Comment (Branchverse) [reply requirements.txt]: validated, are needed

File: docs/rest-api.md
Comment (coffeemakingtoaster) [inline docs/rest-api.md:1]: do we still need this file given we have an openapi? Or is completing the openapi content out of scope for this pr? (which is fine imo, we can always do that lateron)

File: src/rest/app.py [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+    raise HTTPException(status_code=404, detail="Transcription ID not found")
+
+
+@time_it
+@app.post("/transcriptions", dependencies=[Depends(require_api_key)])
+async def post_transcription(
+    file: UploadFile = File(...),
+    language: str | None = Form("en"),
+    settings: str | None = Form('{}'),
+    model: str | None = Form("large-v3"),
```
Comment (coffeemakingtoaster) [inline src/rest/app.py:109]: Is this the default or openapi hints? If its the ladder I think replacing this with tiny would make sense

**Review by coffeemakingtoaster — APPROVED** [review]
LGTM

File: src/rest/app.py (hunk 9 above)
Comment (Branchverse) [reply src/rest/app.py:109]: it is defaults

File: docs/rest-api.md
Comment (Branchverse) [reply docs/rest-api.md:1]: yea later, i would go over redocumentation once we redid websockets aswell, maybe we learn new stuff there and have one work instead of double
